controllers/persistentvolume/reconcile_test.go

```diff
@@ -82,13 +92,14 @@ func TestEnsureExpansionSecret(t *testing.T) {
 		assert.NoError(t, err)
 		assert.Equal(t, reconcile.Result{}, result)
 
-		actual := &corev1.PersistentVolume{}
-		err = reconciler.Client.Get(context.TODO(), request.NamespacedName, actual)
-		assert.NoError(t, err)
-
-		actualSecretRef := actual.Spec.PersistentVolumeSource.CSI.ControllerExpandSecretRef
-		assert.Equal(t, testSC.Parameters[csiExpansionSecretName], actualSecretRef.Name)
-		assert.Equal(t, testSC.Parameters[csiExpansionSecretNamespace], actualSecretRef.Namespace)
+		if c.allowExpansion {
+			actual := &corev1.PersistentVolume{}
+			err = reconciler.Client.Get(context.TODO(), request.NamespacedName, actual)
+			assert.NoError(t, err)
+			actualSecretRef := actual.Spec.PersistentVolumeSource.CSI.ControllerExpandSecretRef
+			assert.Equal(t, testSC.Parameters[csiExpansionSecretName], actualSecretRef.Name)
+			assert.Equal(t, testSC.Parameters[csiExpansionSecretNamespace], actualSecretRef.Namespace)
+		}
```

Comment:
I think in case of false also we should perform some checks like value is not set

Reply:
its an extra check i dont think its necessary,